src/securechip/securechip.c

```diff
@@ -605,7 +605,18 @@ bool securechip_monotonic_increments_remaining(uint32_t* remaining_out)
 
 bool securechip_random(uint8_t* rand_out)
 {
-    return atcab_random(rand_out) == ATCA_SUCCESS;
+    ATCA_STATUS result;
+    uint8_t retries = 0, max_retries = 5;
+    do {
+        result = atcab_random(rand_out);
+    } while (result != ATCA_SUCCESS && retries++ < max_retries);
+    if (result != ATCA_SUCCESS) {
+        char msg[70];
+        sprintf(msg, "securechip_random is not successful with ATCA STATUS %d", result);
+        Abort(msg);
```

Comment:
I wouldn't abort here, as this stops execution of the firmware. If one has to Abort, better do it at the callsite, which is what is happening now already anyway.

Reply:
You're right, but that denies the opportunity of printing the atca status code from the error. Therefore we again can not have an idea about which error we get.

Reply:
If you want to display the error code, then it is best to return it in the function signature and do it at the call site, like how it is done in `securechip_setup()`.

Reply:
I did not want to change function signature for this, therefore I discarded the error message for now. It is a random generator function anyway, I don't think the type of error we get there would help debugging something. Just retrying seems enough for me in that sense, I don't it's worth to change function signature and its calls.